admix/uploader.py

```diff
@@ -85,8 +85,6 @@ def upload(path, rse,
             db.update_data(number, data_dict)
         try:
             clients.upload_client.upload(to_upload)
```

Comment:
Looking at the codes, I wonder what if this line failed for whatever reason? If it failed for network reason, not because of nothing to add, the new try statement you added will still add a rule anyway.

Reply:
but right now honestly I don't have a super good idea how to avoid the boundary cases I described. I feel OK to merge it now actually, if we don't have immediate idea to solve it and write an issue.

Reply:
No, wait. It that line fails, the except that is triggered contains a "return did", so that the function will end and the rule will not be added. Do you agree?

Reply:
Ah sorry I missed that. You are right! I am happy about this PR now.